explainer.md

```diff
@@ -0,0 +1,255 @@
+
+4. When the new page loads, suppress rendering until resources required for first render have been fetched. This state is currently driven by browser heuristics but the eventual goal is to give deterministic control to developers using a [renderblocking](https://github.com/whatwg/html/issues/7131) attribute.
+
+5. Once the page is ready for first render, create the following pseudo elements in the top layer. This is done until the end of rAF callbacks on the page's first rendering lifecycle update[^1] :
+    a. A container and child replaced element for each shared element on the old page using state saved in step 3.
+    b. A container and child replaced element for each element with shared-id attribute on the new page. Only the child element is created if a container with the matching id was already created in step a. above.
+    c. A replaced element for the root/html element identified via ::shared-old(#root).
+
+6. Apply the following UA stylesheet to the pseudo elements on the new page.
+```
+::shared-old(#root), ::shared-new(#root), ::shared-container(#shared-header) {
+  position: fixed;
+  top: 0px;
+  left: 0px;
+}
+
+::shared-old(#root), ::shared-new(#root) {
+  width: 100vw;
+  height: 100vh;
+  content: element(html);
+}
+::shared-old(#root) {
+  // This is the saved output referenced in step 3.
+  content: cached-element(html);
+}
+::shared-new(#root) {
+  content: element(html);
+}
+
+::shared-old(#shared-header), ::shared-new(#shared-header) {
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  
+  content: element(#header);
+}
+::shared-old(#shared-header) {
+  // This is the saved output referenced in step 3.
+  content: cached-element(#header);
+}
+::shared-new(#shared-header) {
+  content: element(#header);
+}
+
+// Update the container's size and transform to the shared element on the new page.
+::shared-container(#shared-header) {
+  width: 200px;
+  height: 200px;
+  transform: translate(0px, 100px);
+}
+```
+
+7. Once no pseudo element has an active animation, remove them from the top layer.
+
+## Live Animatable Properties
+A common capability desirable during transitions is to interpolate styles like border-radius that change the element's shape. The [container transform](https://material.io/design/motion/the-motion-system.html#container-transform) examples show a visual demo of that. Painting properties like the element's border within its image when using the element() function makes this difficult. The following example[^2] shows how developers can control which styles are captured in the element() function vs being applied to the pseudo container on the old page.
+
+```
+<html>
+<head>
+  <style>
+    .animated {
+      width:100px;
+      height:100px;
+      border: 10px solid black;
+      border-radius: 10% 10%;
+      box-shadow: 0px 0px 10px;
+      
+      &:transition {
```

Comment:
Is this SASS? I think this should be pure CSS

Reply:
Was trying to do it without copying the entire declaration. I couldn't find a way to do that with pure CSS.